addons/website_sale/models/res_partner.py

```diff
@@ -49,22 +49,21 @@ def _onchange_property_product_pricelist(self):
                 ),
             }}
 
-    def _can_be_edited_by_current_customer(self, sale_order, address_type):
-        self.ensure_one()
-        children_partner_ids = self.env['res.partner']._search([
-            ('id', 'child_of', sale_order.partner_id.commercial_partner_id.id),
-            ('type', 'in', ('invoice', 'delivery', 'other')),
-        ])
-        if (
-            self == sale_order.partner_id
-            or self.id in children_partner_ids
-        ):
-            # address belongs to the customer
-            if address_type == 'billing':
-                # All addresses are editable as billing
-                return True
-            if address_type == 'delivery' and self.type == 'delivery':
-                # Only delivery addresses are editable as delivery
-                return True
-
-        return False
+    def _can_be_edited_by_current_customer(self, address_type, **kwargs):
+        if sale_order := kwargs.get('order_sudo'):
```

Comment:
Shouldn't the function have `self.ensure_one()` since `sale_order.partner_id` is a Many2one field?
Also, `self.id` would trigger the singleton error in the bad scenario.